src/OrleansRuntime/GrainTypeManager/GrainTypeManager.cs

```diff
@@ -13,18 +15,26 @@ namespace Orleans.Runtime
     internal class GrainTypeManager
     {
         private IDictionary<string, GrainTypeData> grainTypes;
+        private IReadOnlyDictionary<SiloAddress, GrainInterfaceMap> grainInterfaceMapsBySilo;
```

Comment:
What I meant in the other comment was to use `Dictionary<,>` here, since it's cheaper to enumerate:
```
    Method |       Mean |    StdDev |  Gen 0 | Allocated |
---------- |----------- |---------- |------- |---------- |
  Concrete | 27.2424 ns | 0.8356 ns |      - |       0 B |
 Interface | 51.0121 ns | 1.1681 ns | 0.0125 |      56 B |
```